### DISCUSSION TITLE
Count Items Related to Database Item

### DISCUSSION THREAD

**Discussion by perezstudio** [Budibase/budibase#4885] [Help]
I'm using the Budibase DB. I have two tables. One holds the client list, the second holds the orders from the client. I'm trying to count how many orders are currently related to the client and display it on a card. I'm new to Budibase so I'm still trying to make heads or tails of the formula and handlebars system. Help is much appreciated :)

**Comment by mjashanks** [comment]
Hey @Utahbeaner 

You can do this by adding a dynamic formula column, but you will probably need to use javascript to count the items:

1. Add a dynamic formula column
![image](https://user-images.githubusercontent.com/3524181/157842060-3ab27012-3fd5-4485-8d4c-f325d79c7cb2.png)

2. My javascript for counting up "Points" on a relationship called "Kudos"
![image](https://user-images.githubusercontent.com/3524181/157842696-603b1332-ce63-4e48-b1f9-570a7f134bde.png)

**Comment by marco-santulli** [comment]
hi mike,

one question, your example assumes there is a relationship between the table you are managing and the kudos records table right?  I have seen that currently BB creates a new column inside the table to handle relationships, can things be changed so that those correlation columns are located in a metadata area possibly in a separate DB so as not to pollute the master database?

M

**Reply by mjashanks** [reply]
> there is a relationship between the table you are managing and the kudos records table right

Correct.

> I have seen that currently BB creates a new column inside the table to handle relationships

This only happens when using SQL, and when you create a new relationship. When you "Define Existing Relationship", it doesn't create a new column in the database - it just shows a column in Budibase. The column that is shown is just an alias, like the column "UserName" in the query below:

```sql
select
 kudos.*,
 user.name as UserName
from kudos
inner join user on user.id = kudos.userId
```
